Fill.py

```diff
@@ -51,7 +51,10 @@ def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations:
         items_to_place = [items.pop()
                           for items in reachable_items.values() if items]
         for item in items_to_place:
-            item_pool.remove(item)
+            for p, pool_item in enumerate(item_pool):
```

Comment:
this feels super inefficient. Surely we can keep track of the index somehow and pop it from there?

Reply:
Could make items_to_place a list of tuples and build it by enumerating over reachable_items maybe?